stubs/Reflection.phpstub

```diff
@@ -67,10 +430,23 @@ class ReflectionFunction implements Reflector
     public function getAttributes(?string $name = null, int $flags = 0): array {}
 }
 
+class ReflectionFunction extends ReflectionFunctionAbstract
+{
+    /** @psalm-pure */
+    public function __construct(callable $function) {}
```

Comment:
I'm a bit on the fence on this one: not all `callable` types are valid types here.

```php
<?php

class Foo
{
    public function bar() {}
    private function baz() {}
}

var_dump(new ReflectionFunction('Foo::bar'));
var_dump(new ReflectionFunction('Foo::baz'));
```

https://3v4l.org/Fsh3P

It is true that `callable` is a much better type than `non-empty-string|Closure` here, but not all `non-empty-string` types are callable, and not all `callable` are `non-empty-string` (nor valid in this context).

Anyway, this is perhaps just overthinking, and `callable` may be an improvement over the past.

Reply:
However I think we should exclude callable arrays and objects here. `callable-string|Closure` (in docblock) should be better in this context.

Reply:
d9a0cc531113c75ea98dba04cbeccc01f4ae7ba4